src/skin/legacy/skincontext.cpp

```diff
@@ -135,14 +135,19 @@ PixmapSource SkinContext::getPixmapSourceInner(const QString& filename) const {
     return PixmapSource();
 }
 
-QDebug SkinContext::logWarning(const char* file, const int line,
-                               const QDomNode& node) const {
-    return qWarning() << QString("%1:%2 SKIN ERROR at %3:%4 <%5>:")
-                             .arg(file, QString::number(line), m_xmlPath,
-                                  QString::number(node.lineNumber()),
-                                  node.nodeName())
-                             .toUtf8()
-                             .constData();
+QDebug SkinContext::logWarning(const char* file,
+        const int line,
+        const QDomNode& node,
+        const QString& message) const {
+    return qWarning() << QString("Skin parsing failed at %1:%2 <%3>: %4 | %5:%6")
+                                 .arg(m_xmlPath,
+                                         QString::number(node.lineNumber()),
+                                         node.nodeName(),
+                                         message,
+                                         file,
+                                         QString::number(line))
+                                 .toUtf8()
+                                 .constData();
```

Comment:
To allow keep using the stream operators you may inherit from QDebug() 
(Not sure if it is worth the effort) 

Not yet working idea: 
````
class SkinWarning : public QDebug {
  public:
    SkinWarning(const char* file,
        const int line,
        const QDomNode& node,
        const QString& message) 
            : QDebug(QtWarningMsg) {
        *this << "Skin parsing failed at" << QString(m_xmlPath,) + QChar(':') + QString::number(node.lineNumber());
    }

    ~SkinWarning() {
        *this << QString(m_file) + QChar(':') + QString::number(m_line);
    }
  private:
      const char* m_file,
      const int m_line,
};
```

Reply:
What are your plans with this PR?

Reply:
I'll try your suggestions, and if it gives my too much headaches I'd rather merge this as is.

Reply:
aaah, oooh, __FILE__ and __LINE__ are pre-processor macros. A little less magic now.

Anyway, I gave it a shot but it doesn't work as expected, or I just didn't fully understand how you plan to implement. I assumed you aimed to leave all `SKIN_WARNING()` calls unchanged.
https://github.com/ronso0/mixxx/pull/new/skin-warning-class incl. one commit with hotcue 0 to trigger a warning.

Reply:
If you are motivated and manage to fix it, fine. I can re-apply the individual tweaks on top of your branch.
Otherwise let's merge this as is, maybe drop a comment so someone can try again later.